src/util/mp_arith.cpp

```diff
@@ -150,3 +153,142 @@ unsigned integer2unsigned(const BigInt &n)
   assert(ull <= std::numeric_limits<unsigned>::max());
   return (unsigned)ull;
 }
+
+BigInt
+bitwise(const BigInt &a, const BigInt &b, std::function<bool(bool, bool)> f)
+{
+  const auto digits = std::max(a.digits(2), b.digits(2));
+
+  BigInt result = 0;
+  BigInt tmp_a = a, tmp_b = b;
+
+  for(std::size_t i = 0; i < digits; i++)
+  {
+    const bool bit_a = tmp_a.is_odd();
+    const bool bit_b = tmp_b.is_odd();
+    const bool bit_result = f(bit_a, bit_b);
+    if(bit_result)
+      result += power(2, i);
+    tmp_a /= 2;
+    tmp_b /= 2;
+  }
+
+  return result;
+}
+
+BigInt bitwise_or(const BigInt &a, const BigInt &b)
+{
+  (!a.is_negative() && !b.is_negative());
```

Comment:
What is this doing?

Reply:
In CBMC, these are the arguments of a macro called PRECONDITION. 

In ESBMC, they should be assertions.